mlrun/serving/server.py

```diff
@@ -771,7 +769,29 @@ async def run(body):
         model_endpoint_uids=model_endpoint_uids,
     )
 
-    return responses
+    # log the results as artifacts
+    num_of_meps_in_the_graph = len(server.graph.model_endpoints_names)
+    artifact_path = mlrun.mlconf.artifact_path
+    if artifact_path and ("/{{run.uid}}" not in artifact_path):
+        artifact_path = artifact_path.rstrip("/") + "/{{run.uid}}"
+    if num_of_meps_in_the_graph <= 1:
+        context.log_dataset(
+            "prediction", df=pd.DataFrame(responses), artifact_path=artifact_path
```

Comment:
Looking at `extend_artifact_path` and
https://github.com/mlrun/mlrun/blob/db8a1f9777dfb14e4b3b7eb33a051d70032feaa0/mlrun/execution.py#L789
You may just do:
```suggestion
            "prediction", df=pd.DataFrame(responses), artifact_path="+/{{run.uid}}"
```
And add a TODO comment to remove it (with the ticket that @theSaarco opened).